File: 4055-longest-balanced-substring-i/4055-longest-balanced-substring-i.java
class Solution {
    public int longestBalanced(String s) {
        int n = s.length();
        int res = 0;
        for (int i = 0; i < n; i++) {
            int freq[] = new int[26];
            for (int j = i; j < n; j++) {
                freq[s.charAt(j) - 'a']++;
                int flag = 1;
                int maxi = 0;
                for (int ele : freq) 
                    maxi = Math.max(maxi, ele);
                for (int ele : freq) {
                    if (ele > 0 && ele != maxi)
                        flag = 0; 
                }
                if (flag == 1)
                    res = Math.max(res, j - i + 1);
            }
        }
        return res; 
    }
}

File: 4055-longest-balanced-substring-i/4055-longest-balanced-substring-i.py
class Solution:
    def longestBalanced(self, s: str) -> int:
        n = len(s)
        maxi = 0
        for i in range(0, n):
            count = dict()
            for j in range(i, n):
                count[s[j]] = count.get(s[j], 0) + 1
                st = set(list(count.values()))
                if len(st) == 1:
                    maxi = max(maxi, j - i + 1)

        return maxi

